aptos-move/framework/aptos-stdlib/sources/signature.move

```diff
@@ -177,6 +184,14 @@ module aptos_std::signature {
         signature: vector<u8>
     ): (vector<u8>, bool);
 
+    public fun ed25519_verify_t<T: drop> (signature: vector<u8>, public_key: vector<u8>, data: T): bool {
+        let encoded = Proof {
+            type_info: type_info::type_of<T>(),
+            inner: data,
+        };
+        ed25519_verify(signature, public_key, bcs::to_bytes(&encoded))
```

Comment:
So the wallet code will have to BCS encode this struct when sending a key-rotation transaction.

I assume that's not a problem.